### ISSUE TITLE
[Bug]: useServerHandoff fails with multiple SSR React instances

### ISSUE THREAD

**Issue by thebuilder** [tailwindlabs/headlessui#701]
### What package within Headless UI are you using?

@headlessui/react

### What version of that package are you using?

v1.3.0

### What browser are you using?

N/A

### Reproduction repository

None

### Describe your issue

I've run into an issue when we have multiple smaller React instances on our site that contain a HeadlessUI component. 
During hydration the first component will set the `serverHandoffComplete` as `true`, causing the remaining `ReactDOM.hydrate()` calls to get hydration issues. 
The result is that the `id` of HeadlessUI components will still contain `undefined`, since React skips changing the element.

This might not be the most common approach to using React, but is still a perfectly valid approach. 😄

The simplest solution, and what I've done as a temporary fix, is to allow us to set the value of `serverHandoffComplete`:

```ts
export function setHandoffComplete(handoff: boolean) {
  state.serverHandoffComplete = handoff;
}
```

This will allow me to reset the handoff after each module is hydrated, until the entire page has been fully hydrated.
This then also needs to be exported in the `index.ts`, so it can be accessed when consuming the module.

I can make a PR for it.

Just to give an idea of how I'm consuming it:

```tsx
import { setHandoffComplete } from '@headlessui/react';

export async function hydratePage() {
  const modules = Array.from(
    document.querySelectorAll<HTMLElement>('[data-module]')
  );

  await Promise.all(
    modules.map(async (el) => {
      const name = el.getAttribute('data-module') as ModuleTypes;
      const script = el.nextElementSibling as HTMLScriptElement;
      const props = JSON.parse(script?.textContent ?? '{}');

      if (el) await hydrateModule(el, name, props);
      setHandoffComplete(false);
    })
  );
  
  setHandoffComplete(true);
}
```

**Comment by thebuilder** [comment]
The server handoff is a good idea, and normally it would work just fine - But without this escape hatch, we can't really use library. This issue also exists with similar solutions, like Reach UI.

**Comment by thebuilder** [comment]
Resolved in 1.5
